lib/services/sf.js

```diff
@@ -623,6 +624,18 @@ function StateAbstract(options) {
             // success flag is false, the operation we tried to perform failed
             if (body && !body.success) {
               const data = body.data;
+              const auth = getCurrentAuth();
+
+              if (body.code === GSErrors.code.ID_TOKEN_INVALID && data.authnMethod === 'TOKEN') {
+                Logger.getInstance().debug('ID Token being used has expired. Reauthenticating');
+                const key = Util.buildCredentialCacheKey(connectionConfig.account, 
+                  connectionConfig.username, authenticationTypes.ID_TOKEN_AUTHENTICATOR);
+                await GlobalConfig.getCredentialManager().remove(key);
```

Comment:
are the called method async? I think I saw it without async and with `fs.xyzSync` methods used?

Reply:
Yes, I created the JSON credential manager with sync methods. However, we need to consider a scenario when users use the async functions for the custom credential manager. That is why I added await or then keywords.